apps/sms/views/shared/js/activity_handler.js

```diff
@@ -438,13 +441,19 @@ var ActivityHandler = {
                 text = slideArray[i].text;
                 break;
               }
-              text = text ? text : _('mms-message');
-              callback(text);
+
+              if (text) {
+                defer.resolve(text);
+              } else {
+                formatValue('mms-message').then(defer.resolve);
+              }
             });
           }
+
+          return defer.promise;
         }
 
-        Contacts.findByAddress(message.sender).then(function(contacts) {
+        return Contacts.findByAddress(message.sender).then(function(contacts) {
```

Comment:
Since we return Promise here now, I'm wondering if we should now use ```mozSetMessageHandlerPromise``` instead?

[1] http://mxr.mozilla.org/mozilla-central/source/dom/webidl/Navigator.webidl#301